script/ZenithL2.s.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.26;
+
+// system contracts
+import {RollupOrders} from "../src/orders/RollupOrders.sol";
+import {RollupPassage} from "../src/passage/RollupPassage.sol";
+// simple erc20
+import {SimpleERC20} from "simple-erc20/SimpleERC20.sol";
+// utils
+import {Script, console2} from "forge-std/Script.sol";
+
+// create2 address for Permit2
+address constant PERMIT2 = 0x000000000022D473030F116dDEE9F6B43aC78BA3;
+// address of Minter with permission to mint ERC20 tokens
+address constant MINTER = 0x9999999999999999999999999999999999999999;
+
+contract L2Script is Script {
+    // deploy:
+    // forge script L2Script --sig "deploySystem()" --rpc-url $RPC_URL --broadcast [signing args]
+    function deploySystem() public returns (address rollupPassage, address rollupOrders, address wbtc, address usdt) {
+        vm.startBroadcast();
+
+        // deploy system contracts
+        rollupPassage = address(new RollupPassage{salt: "zenith.rollupPassage"}(PERMIT2));
```

Comment:
do we want to do salt-mining to find low addresses?

Reply:
hmm. well, we don't technically have to. we could handpick better addrs than we could reasonably mine by simply directly editing the `allocs` file. though, that would mean generating the allocs would be less reproducible.

Reply:
yeah, we'd also have to do some find-and-replace shenangians to update immutables and storage vars. which would be annoying. but it would be awful nice to have them have short addresses 🤔

Reply:
depends on how dearly we wish for the script to produce the exact final allocs 

if we want efficient addrs for all contracts (&& fully deterministic script -> allocs) we might want to consider forking the USDC deploy script and making them CREATE2 deploys, because rn they're just CREATE and can't be mined

Reply:
> yeah, we'd also have to do some find-and-replace shenangians to update immutables and storage vars. which would be annoying. but it would be awful nice to have them have short addresses 🤔

well, actually, out of the following contracts
- permit2 
- RollupPassage
- RollupOrders
- WBTC
- USDT
- SafeProxyFactory
- SafeL2
- CompatibilityFallbackHandler

the following is true,
- permit2 is the only one that's stored as an immutable (in RollupPassage and RollupOrders) - BUT i think it's better for permit2 to remain at the same address it's at on mainnet
- SafeL2 and CompatibilityFallbackHandler are stored in the storage slots of the Safe instance we deploy for USDC admin, but we can actually just deploy the USDC admin passing whatever pegged addresses we choose
- the rest of the listed contracts are not present in any immutables or storage slots 

so we could actually easily changethe addrs without find/replacing anything in code or storage.

not listed in the bullets above are 
- the USDC admin Safe instance - i think we're not worried about cheaper calls to this one
- the entire USDC protocol - for which, yes, we'd have to do a lot of find-and-replace